File: java/Samples/BiometricStandards/VB/FCRecordSampleVB/RawFaceImageOptionsForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Windows.Forms
Imports Neurotec.Biometrics.Standards

Partial Public Class RawFaceImageOptionsForm
	Inherits AddFaceImageForm
	Public Enum RawFaceImageOptionsFormMode
		Load = 1
		Save = 2
	End Enum

	Public Sub New()
		InitializeComponent()

		cbImageColorSpace.Items.AddRange(System.Enum.GetNames(GetType(FcrImageColorSpace)))
		cbImageColorSpace.SelectedIndex = 0
	End Sub

	Private _mode As RawFaceImageOptionsFormMode = RawFaceImageOptionsFormMode.Load
	Public Property Mode() As RawFaceImageOptionsFormMode
		Get
			Return _mode
		End Get
		Set(ByVal value As RawFaceImageOptionsFormMode)
			_mode = value
			OnModeChanged()
		End Set
	End Property

	Public Property ImageColorSpace() As FcrImageColorSpace
		Get
			Return CType(System.Enum.Parse(GetType(FcrImageColorSpace), cbImageColorSpace.SelectedItem.ToString()), FcrImageColorSpace)
		End Get
		Set(ByVal value As FcrImageColorSpace)
			cbImageColorSpace.SelectedItem = System.Enum.GetName(GetType(FcrImageColorSpace), value)
		End Set
	End Property

	Public Property ImageWidth() As UShort
		Get
			Return UShort.Parse(tbWidth.Text)
		End Get
		Set(ByVal value As UShort)
			tbWidth.Text = value.ToString()
		End Set
	End Property

	Public Property ImageHeight() As UShort
		Get
			Return UShort.Parse(tbHeight.Text)
		End Get
		Set(ByVal value As UShort)
			tbHeight.Text = value.ToString()
		End Set
	End Property

	Public ReadOnly Property VendorColorSpace() As Byte
		Get
			Return Byte.Parse(tbVendorImageColorSpace.Text)
		End Get
	End Property

	Private Sub cbImageColorSpace_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbImageColorSpace.SelectedIndexChanged
		If CType(System.Enum.Parse(GetType(FcrImageColorSpace), cbImageColorSpace.SelectedItem.ToString()), FcrImageColorSpace) = FcrImageColorSpace.Vendor Then
			If (Not tbVendorImageColorSpace.Enabled) Then
				tbVendorImageColorSpace.Enabled = True
			End If
		Else
			tbVendorImageColorSpace.Enabled = False
			tbVendorImageColorSpace.Text = "0"
		End If
	End Sub

	Private Sub tbWidthHeight_Validating(ByVal sender As Object, ByVal e As CancelEventArgs) Handles tbWidth.Validating, tbHeight.Validating
		Dim s As Control = TryCast(sender, Control)
		If s Is Nothing Then
			Return
		End If

		Dim result As UShort
		If UShort.TryParse(s.Text, result) Then
			errorProvider.SetError(s, String.Empty)
		Else
			errorProvider.SetError(s, "Incorrect data entered. Data should be in range 0 to 65535.")
			e.Cancel = True
		End If
	End Sub

	Private Sub tbVendorImageColorSpace_Validating(ByVal sender As Object, ByVal e As CancelEventArgs) Handles tbVendorImageColorSpace.Validating
		Dim s As Control = TryCast(sender, Control)
		If s Is Nothing Then
			Return
		End If

		Dim result As Byte
		If Byte.TryParse(s.Text, result) Then
			errorProvider.SetError(s, String.Empty)
		Else
			errorProvider.SetError(s, "Incorrect data entered. Data should be in range 0 to 255.")
			e.Cancel = True
		End If
	End Sub

	Private Sub OnModeChanged()
		Select Case Mode
			Case RawFaceImageOptionsFormMode.Load
				Text = "Add face from data"
			Case RawFaceImageOptionsFormMode.Save
				Text = "Save face as data"
		End Select
	End Sub
End Class

File: java/Samples/BiometricStandards/CS/FCRecordSampleCS/RawFaceImageOptionsForm.cs
using System;
using System.ComponentModel;
using System.Windows.Forms;
using Neurotec.Biometrics.Standards;

namespace Neurotec.Samples
{
	public partial class RawFaceImageOptionsForm : AddFaceImageForm
	{
		public enum RawFaceImageOptionsFormMode
		{
			Load = 1,
			Save = 2
		}

		public RawFaceImageOptionsForm()
		{
			InitializeComponent();

			cbImageColorSpace.Items.AddRange(Enum.GetNames(typeof(FcrImageColorSpace)));
			cbImageColorSpace.SelectedIndex = 0;
		}

		private RawFaceImageOptionsFormMode _mode = RawFaceImageOptionsFormMode.Load;
		public RawFaceImageOptionsFormMode Mode
		{
			get
			{
				return _mode;
			}
			set
			{
				_mode = value;
				OnModeChanged();
			}
		}

		public FcrImageColorSpace ImageColorSpace
		{
			get
			{
				return (FcrImageColorSpace)Enum.Parse(typeof(FcrImageColorSpace), cbImageColorSpace.SelectedItem.ToString());
			}
			set
			{
				cbImageColorSpace.SelectedItem = Enum.GetName(typeof(FcrImageColorSpace), value);
			}
		}

		public ushort ImageWidth
		{
			get
			{
				return ushort.Parse(tbWidth.Text);
			}
			set
			{
				tbWidth.Text = value.ToString();
			}
		}

		public ushort ImageHeight
		{
			get
			{
				return ushort.Parse(tbHeight.Text);
			}
			set
			{
				tbHeight.Text = value.ToString();
			}
		}

		public byte VendorColorSpace
		{
			get
			{
				return byte.Parse(tbVendorImageColorSpace.Text);
			}
		}

		private void cbImageColorSpace_SelectedIndexChanged(object sender, EventArgs e)
		{
			if ((FcrImageColorSpace)Enum.Parse(typeof(FcrImageColorSpace), cbImageColorSpace.SelectedItem.ToString()) == FcrImageColorSpace.Vendor)
			{
				if (!tbVendorImageColorSpace.Enabled)
					tbVendorImageColorSpace.Enabled = true;
			}
			else
			{
				tbVendorImageColorSpace.Enabled = false;
				tbVendorImageColorSpace.Text = @"0";
			}
		}

		private void tbWidthHeight_Validating(object sender, CancelEventArgs e)
		{
			Control s = sender as Control;
			if (s == null) return;

			ushort result;
			if (ushort.TryParse(s.Text, out result))
			{
				errorProvider.SetError(s, string.Empty);
			}
			else
			{
				errorProvider.SetError(s, "Incorrect data entered. Data should be in range 0 to 65535.");
				e.Cancel = true;
			}
		}

		private void tbVendorImageColorSpace_Validating(object sender, CancelEventArgs e)
		{
			Control s = sender as Control;
			if (s == null) return;

			byte result;
			if (byte.TryParse(s.Text, out result))
			{
				errorProvider.SetError(s, string.Empty);
			}
			else
			{
				errorProvider.SetError(s, "Incorrect data entered. Data should be in range 0 to 255.");
				e.Cancel = true;
			}
		}

		private void OnModeChanged()
		{
			switch (Mode)
			{
				case RawFaceImageOptionsFormMode.Load:
					Text = @"Add face from data";
					break;
				case RawFaceImageOptionsFormMode.Save:
					Text = @"Save face as data";
					break;
			}
		}
	}
}
